models/classes/runner/time/QtiTimeLine.php

```diff
@@ -0,0 +1,306 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien.conan@vesperiagroup.com>
+ */
+
+namespace oat\taoQtiTest\models\runner\time;
+
+use oat\taoTests\models\runner\time\IncompleteRangeException;
+use oat\taoTests\models\runner\time\InconsistentRangeException;
+use oat\taoTests\models\runner\time\MalformedRangeException;
+use oat\taoTests\models\runner\time\TimeException;
+use oat\taoTests\models\runner\time\TimeLine;
+use oat\taoTests\models\runner\time\TimePoint;
+
+/**
+ * Class QtiTimeLine
+ * @package oat\taoQtiTest\models\runner\time
+ */
+class QtiTimeLine implements TimeLine
+{
+    /**
+     * The list of TimePoint representing the TimeLine
+     * @var array
+     */
+    protected $points = [];
+
+    /**
+     * QtiTimeLine constructor.
+     * @param array $points
+     */
+    public function __construct($points = null)
+    {
+        if (isset($points)) {
+            foreach ($points as $point) {
+                $this->add($point);
+            }
+        }
+    }
+
+    /**
+     * String representation of object
+     * @link http://php.net/manual/en/serializable.serialize.php
+     * @return string the string representation of the object or null
+     * @since 5.1.0
+     */
+    public function serialize()
+    {
+        $data = [];
+
+        foreach ($this->points as $point) {
+            $data [] = serialize($point);
+        }
+
+        return serialize($data);
```

Comment:
Why do you iterate over the points ? Isn't `serialize($this->points)` sufficient ?